qupath-core/src/main/java/qupath/lib/objects/utils/ObjectMerger.java

```diff
@@ -570,6 +680,74 @@ private static PathObject mergeObjects(List<? extends PathObject> pathObjects) {
         return mergedObject;
     }
 
+    private static MeasurementList mergeMeasurements(List<? extends PathObject> pathObjects, MeasurementStrategy measurementStrategy) {
+        if (pathObjects.isEmpty()) {
+            return null;
+        }
+        return switch(measurementStrategy) {
+            case IGNORE -> null;
+            case USE_FIRST -> pathObjects.getFirst().getMeasurementList();
+            case MEAN -> {
+                var out = MeasurementListFactory.createMeasurementList(
+                        pathObjects.getFirst().getMeasurementList().size(),
+                        MeasurementList.MeasurementListType.FLOAT
+                );
+                for (var es: pathObjects.getFirst().getMeasurements().entrySet()) {
+                    out.put(
+                            es.getKey(),
+                            pathObjects.stream().mapToDouble(po -> po.getMeasurementList().get(es.getKey())).sum() / pathObjects.size());
+                }
+                yield out;
+            }
+            case WEIGHTED_MEAN -> {
+                var out = MeasurementListFactory.createMeasurementList(
+                        pathObjects.getFirst().getMeasurementList().size(),
+                        MeasurementList.MeasurementListType.FLOAT
+                );
+                Map<ROI, Double> areas = new HashMap<>();
```

Comment:
Feels a bit like the stream addiction is strong here, and readability is reduced... I'd tend towards having a separate function
```java
private static Map<PathObject, Double> computeWeights(Collection<? extends PathObject> pathObjects)
```
that computes weights that sum to 1 for each object, returning these in a map. That would also have the benefit of meaning that you could check if the area sum is zero, and either resort to length/point count or just log a warning (if we can assume that the code is unlikely to ever be merging lines or points).